diff --git a/node_modules/@changesets/get-github-info/dist/changesets-get-github-info.cjs.js b/node_modules/@changesets/get-github-info/dist/changesets-get-github-info.cjs.js
index 4187b5c..06f8653 100644
--- a/node_modules/@changesets/get-github-info/dist/changesets-get-github-info.cjs.js
+++ b/node_modules/@changesets/get-github-info/dist/changesets-get-github-info.cjs.js
@@ -205,7 +205,7 @@ const GHDataLoader = new DataLoader__default["default"](async requests => {
       data = _objectWithoutProperties(_ref2, _excluded2);
     return cleanedData[repo][data.kind][data.kind === "pull" ? data.pull : data.commit];
   });
-});
+}, { maxBatchSize: 100 });
 async function getInfo(request) {
   if (!request.commit) {
     throw new Error("Please pass a commit SHA to getInfo");
