diff --git a/node_modules/@graphql-codegen/visitor-plugin-common/index.cjs.js b/node_modules/@graphql-codegen/visitor-plugin-common/index.cjs.js
index afec571..ff0ed2f 100644
--- a/node_modules/@graphql-codegen/visitor-plugin-common/index.cjs.js
+++ b/node_modules/@graphql-codegen/visitor-plugin-common/index.cjs.js
@@ -2340,7 +2340,7 @@ class SelectionSetToObject {
                     this._appendToTypeMap(types, typeOnSchema.name, spreadsUsage[typeOnSchema.name]);
                     this._collectInlineFragments(typeOnSchema, inlines, types);
                 }
-                else if (graphql.isInterfaceType(typeOnSchema) && parentType.isTypeOf(typeOnSchema, null, null)) {
+                else if (graphql.isInterfaceType(typeOnSchema) && parentType.getInterfaces().includes(typeOnSchema)) {
                     this._appendToTypeMap(types, parentType.name, fields);
                     this._appendToTypeMap(types, parentType.name, spreadsUsage[parentType.name]);
                     this._collectInlineFragments(typeOnSchema, inlines, types);
