diff --git a/node_modules/@graphql-mesh/apollo-link/index.mjs b/node_modules/@graphql-mesh/apollo-link/index.mjs
index d1e2b34..fbf27e3 100644
--- a/node_modules/@graphql-mesh/apollo-link/index.mjs
+++ b/node_modules/@graphql-mesh/apollo-link/index.mjs
@@ -1,6 +1,8 @@
-import { ApolloLink, Observable } from '@apollo/client/core';
 import { isAsyncIterable } from '@graphql-tools/utils';
 import { getOperationAST } from 'graphql';
+import * as apolloImport from '@apollo/client';
+
+const apollo = apolloImport?.default ?? apolloImport;
 
 const ROOT_VALUE = {};
 function createMeshApolloRequestHandler(options) {
@@ -10,7 +12,7 @@ function createMeshApolloRequestHandler(options) {
             throw new Error('GraphQL operation not found');
         }
         const operationFn = operationAst.operation === 'subscription' ? options.subscribe : options.execute;
-        return new Observable(observer => {
+        return new apollo.Observable(observer => {
             Promise.resolve()
                 .then(async () => {
                 const results = await operationFn(operation.query, operation.variables, operation.getContext(), ROOT_VALUE, operation.operationName);
@@ -38,7 +40,7 @@ function createMeshApolloRequestHandler(options) {
         });
     };
 }
-class MeshApolloLink extends ApolloLink {
+class MeshApolloLink extends apollo.ApolloLink {
     constructor(options) {
         super(createMeshApolloRequestHandler(options));
     }
