diff --git a/node_modules/@graphql-mesh/http/esm/index.js b/node_modules/@graphql-mesh/http/esm/index.js
index 5e35727..40fc28a 100644
--- a/node_modules/@graphql-mesh/http/esm/index.js
+++ b/node_modules/@graphql-mesh/http/esm/index.js
@@ -1,4 +1,4 @@
-import { fs, path } from '@graphql-mesh/cross-helpers';
+import { fs, path } from '@graphql-mesh/cross-helpers/browser';
 import { DefaultLogger, pathExists, withCookies } from '@graphql-mesh/utils';
 import { createRouter, Response } from '@whatwg-node/router';
 import { graphqlHandler } from './graphqlHandler.js';
