diff --git a/node_modules/@lerna/conventional-commits/lib/update-changelog.js b/node_modules/@lerna/conventional-commits/lib/update-changelog.js
index dfe5030..dd6c650 100644
--- a/node_modules/@lerna/conventional-commits/lib/update-changelog.js
+++ b/node_modules/@lerna/conventional-commits/lib/update-changelog.js
@@ -68,6 +68,9 @@ function updateChangelog(pkg, type, { changelogPreset, rootPath, tagPrefix = "v"
     ]).then(([newEntry, [changelogFileLoc, changelogContents]]) => {
       log.silly(type, "writing new entry: %j", newEntry);
 
+      if (newEntry.includes('**Note:** Version bump only for package')) {
+        return { logPath: changelogFileLoc }
+      }
       const content = [CHANGELOG_HEADER, newEntry, changelogContents].join(BLANK_LINE);
 
       return fs.writeFile(changelogFileLoc, content.trim() + EOL).then(() => {
