diff --git a/node_modules/@lerna/version/index.js b/node_modules/@lerna/version/index.js
index 3fcea7c..1319717 100644
--- a/node_modules/@lerna/version/index.js
+++ b/node_modules/@lerna/version/index.js
@@ -558,7 +558,7 @@ class VersionCommand extends Command {
           changedFiles.add(logPath);
 
           // add release notes
-          if (independentVersions) {
+          if (independentVersions && newEntry) {
             this.releaseNotes.push({
               name: pkg.name,
               notes: newEntry,
diff --git a/node_modules/@lerna/version/lib/create-release.js b/node_modules/@lerna/version/lib/create-release.js
index be433d7..004618e 100644
--- a/node_modules/@lerna/version/lib/create-release.js
+++ b/node_modules/@lerna/version/lib/create-release.js
@@ -37,7 +37,7 @@ function createRelease(client, { tags, releaseNotes }, { gitRemote, execOpts })
       const tag = name === "fixed" ? tags[0] : tags.find((t) => t.startsWith(`${name}@`));
 
       /* istanbul ignore if */
-      if (!tag) {
+      if (!tag || !notes) {
         return Promise.resolve();
       }
 
