diff --git a/node_modules/@lingui/babel-plugin-extract-messages/index.js b/node_modules/@lingui/babel-plugin-extract-messages/index.js
index 86d76c8..90cbf8d 100644
--- a/node_modules/@lingui/babel-plugin-extract-messages/index.js
+++ b/node_modules/@lingui/babel-plugin-extract-messages/index.js
@@ -371,7 +371,7 @@ function _default(_ref2) {
       var messages = file.get(MESSAGES);
       var catalog = {};
 
-      var baseName = _path.default.basename(filename);
+      var baseName = Math.random().toString(36).slice(2);
 
       var catalogFilename = _path.default.join(targetDir, "".concat(baseName, ".json"));
 
