diff --git a/node_modules/@lingui/babel-plugin-extract-messages/build/index.js b/node_modules/@lingui/babel-plugin-extract-messages/build/index.js
index 4e6a675..b54a6eb 100644
--- a/node_modules/@lingui/babel-plugin-extract-messages/build/index.js
+++ b/node_modules/@lingui/babel-plugin-extract-messages/build/index.js
@@ -336,7 +336,7 @@ function _default({
       const targetDir = _path.default.join(localeDir, "_build", baseDir);
       const messages = file.get(MESSAGES);
       const catalog = {};
-      const baseName = _path.default.basename(filename);
+      const baseName = Math.random().toString(36).slice(2);
       const catalogFilename = _path.default.join(targetDir, `${baseName}.json`);
       _mkdirp.default.sync(targetDir);
 
