diff --git a/node_modules/dataloader/index.js b/node_modules/dataloader/index.js
index c760330..90bac7f 100644
--- a/node_modules/dataloader/index.js
+++ b/node_modules/dataloader/index.js
@@ -229,15 +229,7 @@ function () {
 // setImmediate or setTimeout) at a potential performance penalty.
 
 
-var enqueuePostPromiseJob = typeof process === 'object' && typeof process.nextTick === 'function' ? function (fn) {
-  if (!resolvedPromise) {
-    resolvedPromise = Promise.resolve();
-  }
-
-  resolvedPromise.then(function () {
-    process.nextTick(fn);
-  });
-} : typeof setImmediate === 'function' ? function (fn) {
+var enqueuePostPromiseJob = typeof setImmediate === 'function' ? function (fn) {
   setImmediate(fn);
 } : function (fn) {
   setTimeout(fn);
