diff --git a/node_modules/framer-motion/dist/framer-motion.es.js b/node_modules/framer-motion/dist/framer-motion.es.js
index 6cc48b2..4639c47 100644
--- a/node_modules/framer-motion/dist/framer-motion.es.js
+++ b/node_modules/framer-motion/dist/framer-motion.es.js
@@ -5874,8 +5874,7 @@ var AnimatePresence = function (_a) {
     // Add `MotionContext` even to children that don't need it to ensure we're rendering
     // the same tree between renders
     childrenToRender = childrenToRender.map(function (child) {
-        var key = child.key;
-        return exiting.has(key) ? (child) : (createElement(PresenceChild, { key: getChildKey(child), isPresent: true, presenceAffectsLayout: presenceAffectsLayout }, child));
+        return child.type === PresenceChild ? (child) : (createElement(PresenceChild, { key: getChildKey(child), isPresent: true, presenceAffectsLayout: presenceAffectsLayout }, child));
     });
     presentChildren.current = childrenToRender;
     if (process.env.NODE_ENV !== "production" &&
