diff --git a/node_modules/lodash.get/index.js b/node_modules/lodash.get/index.js
index 0eaadec..689a021 100644
--- a/node_modules/lodash.get/index.js
+++ b/node_modules/lodash.get/index.js
@@ -46,7 +46,7 @@ var freeGlobal = typeof global == 'object' && global && global.Object === Object
 var freeSelf = typeof self == 'object' && self && self.Object === Object && self;
 
 /** Used as a reference to the global object. */
-var root = freeGlobal || freeSelf || Function('return this')();
+var root = freeGlobal || freeSelf || globalThis;
 
 /**
  * Gets the value at `key` of `object`.
