diff --git a/node_modules/lodash.set/index.js b/node_modules/lodash.set/index.js
index 9f3ed6b..410b37d 100644
--- a/node_modules/lodash.set/index.js
+++ b/node_modules/lodash.set/index.js
@@ -50,7 +50,7 @@ var freeGlobal = typeof global == 'object' && global && global.Object === Object
 var freeSelf = typeof self == 'object' && self && self.Object === Object && self;
 
 /** Used as a reference to the global object. */
-var root = freeGlobal || freeSelf || Function('return this')();
+var root = freeGlobal || freeSelf || globalThis;
 
 /**
  * Gets the value at `key` of `object`.
