diff --git a/node_modules/next/dist/build/swc/options.js b/node_modules/next/dist/build/swc/options.js
index 28b051e..7bc0890 100755
--- a/node_modules/next/dist/build/swc/options.js
+++ b/node_modules/next/dist/build/swc/options.js
@@ -181,9 +181,12 @@ function getLoaderSWCOptions({ filename , development , isServer , pagesDir , is
             // Ensure Next.js internals are output as commonjs modules
             ...isNextDist ? {
                 module: {
-                    type: 'commonjs'
+                    type: 'es6'
                 }
             } : {},
+            "env": {
+                "targets": "> 1% in alt-EU, not ie <= 11"
+            },
             disableNextSsg: !isPageFile,
             isDevelopment: development,
             isServer,
