diff --git a/node_modules/next/dist/build/swc/options.js b/node_modules/next/dist/build/swc/options.js
index bc39fc8..c269f12 100644
--- a/node_modules/next/dist/build/swc/options.js
+++ b/node_modules/next/dist/build/swc/options.js
@@ -178,9 +178,12 @@ function getLoaderSWCOptions({ filename , development , isServer , pagesDir , is
             // Ensure Next.js internals are output as commonjs modules
             ...isNextDist ? {
                 module: {
-                    type: 'commonjs'
+                    type: 'es6'
                 }
             } : {},
+            "env": {
+                "targets": "> 1% in alt-EU, not ie <= 11"
+            },
             disableNextSsg: !isPageFile,
             isDevelopment: development,
             isServer,
