diff --git a/node_modules/framer-motion/dist/es/value/index.js b/node_modules/framer-motion/dist/es/value/index.js
index 4378a9b..a4aae7d 100644
--- a/node_modules/framer-motion/dist/es/value/index.js
+++ b/node_modules/framer-motion/dist/es/value/index.js
@@ -68,7 +68,7 @@ var MotionValue = /** @class */ (function () {
             if (_this.lastUpdated !== timestamp) {
                 _this.timeDelta = delta;
                 _this.lastUpdated = timestamp;
-                sync.postRender(_this.scheduleVelocityCheck);
+                // sync.postRender(_this.scheduleVelocityCheck);
             }
             // Update update subscribers
             if (_this.prev !== _this.current) {
